### DISCUSSION TITLE
Model validations in rodauth-rails ?

### DISCUSSION THREAD

**Discussion by jairfg** [janko/rodauth-rails#248] [Q&A]
Hello community, I am implementing Rodauth-Rails in a project where I already had an account model and had validations, but when implementing Rodauth I apparently skip these validations when creating an account with /create-account.

This is part of the code, when creating a new account, the validations are not executed and it never goes through "restrict_user_limits",

![Captura desde 2023-11-25 10-57-28](https://github.com/janko/rodauth-rails/assets/58352131/e2df872a-dc4e-40e6-bc03-4a8219a68550)

**Comment by janko** [comment]
Rodauth doesn't use Active Record models, it only uses Active Record's DB connection with Sequel. You could manually run model validations in a `before_create_account` hook and throw them in Rodauth.

I don't see the validation, so I don't know if it adds errors on `:base` or a field that's present in the registration form. But let's say you want to validate presence of a custom `name` field (you can see how one is configured in the official Rails demo):

```rb
class User < ApplicationRecord
  validates_presence_of :name
end
```
```rb
class RodauthMain < Rodauth::Rails::Auth
  configure do
    before_create_account do
      rails_account.assign_attributes(name: param("name"))

      if rails_account.invalid?
        rails_account.errors.to_hash.each do |name, errors|
          throw_error_status(422, name.to_s, errors.first)
        end
      end
    end
  end
end
```

**Reply by jairfg** [reply]
@janko  thank you very much for your reply, I decided to separate and create a model Account and relate it to the User model.  The Account model has only login fields and the User model has all the other important fields.
